tools/mo/openvino/tools/mo/main.py

```diff
@@ -329,11 +376,9 @@ def check_fallback(argv : argparse.Namespace):
     if argv.use_new_frontend:
         return fallback_reasons
 
-    fallback_reasons['extensions'] = \
-        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
-            and argv.extensions != import_extensions.default_path() # extensions arg has default value
-    fallback_reasons['transformations_config'] = \
-        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
+    fallback_reasons['extensions'] = legacy_extensions_used
+    fallback_reasons['transformations_config'] = legacy_transformations_config_used
+    fallback_reasons['input_freezing'] = input_freezig_used
```

Comment:
please note that different frontends can have different fallback reasons based on their limitations. For example, TF frontend is able to freeze input. Please do not add `input_freezing` reason in case TF

Reply:
I see, but `check_fallback` is called only if `moc_front_end` is set. TF still uses the legacy path as default so it can't be executed for TF